src/components/organisms/connections/integrations/asana/edit.tsx

```diff
@@ -3,25 +3,39 @@ import React, { useState } from "react";
 import { useTranslation } from "react-i18next";
 import { Link } from "react-router-dom";
 
+import { DeploymentStateVariant } from "@src/enums";
+import { ModalName } from "@src/enums/components";
 import { useConnectionForm } from "@src/hooks";
+import { useCacheStore, useModalStore } from "@src/store";
 import { asanaIntegrationSchema } from "@validations";
 
 import { Button, ErrorMessage, SecretInput, Spinner } from "@components/atoms";
 import { Accordion } from "@components/molecules";
+import { WarningDeploymentActivetedModal } from "@components/organisms";
 
 import { ExternalLinkIcon, FloppyDiskIcon } from "@assets/image/icons";
 
 export const AsanaIntegrationEditForm = () => {
 	const { t } = useTranslation("integrations");
 	const [lockState, setLockState] = useState(true);
-
+	const { deployments } = useCacheStore();
+	const { openModal } = useModalStore();
 	const { errors, handleSubmit, isLoading, onSubmitEdit, register, setValue } = useConnectionForm(
 		asanaIntegrationSchema,
 		"edit"
 	);
 
+	const handleFormSubmit = () => {
+		if (deployments?.length && deployments[0].state === DeploymentStateVariant.active) {
```

Comment:
Don't check it each time in each component, but do something like this:
```
const initialState: Pick<
    CacheStore,
    "loading" | "deployments" | "triggers" | "variables" | "events" | "envId" | "currentProjectId" | "hasActiveDeployments"
> = {
    loading: {
        deployments: false,
        triggers: false,
        variables: false,
        events: false,
    },
    deployments: undefined,
    variables: [],
    triggers: [],
    events: undefined,
    currentProjectId: undefined,
    envId: undefined,
    hasActiveDeployments: false,
};

const store: StateCreator<CacheStore> = (set, get) => ({
    ...initialState,

    get hasActiveDeployments() {
        const { deployments } = get();
        return Boolean(
            deployments?.length && 
            deployments.some(deployment => deployment.state === DeploymentStateVariant.active)
        );
    },

```

And then use `hasActiveDeployments` as a getter in each component.

Reply:
added and refactored